### ISSUE TITLE
SC2086: backticks quoting gives wrong warning

### ISSUE THREAD

**Issue by mulle-nat** [koalaman/shellcheck#478]
```shell
xxx="xxx/yyy.bar .bar" ; echo "`basename \"${xxx}\"`"
```
gives 

```console
SC2086: Double quote to prevent globbing and word splitting.
```

Which is IMO wrong, the variable does not glob or word split. The output `yyy.bar .bar` is as expected (sh). And no I don't want to convert off backticks, thank you :)

**Comment by bittorf** [comment]
IMHO this is a useless use of 'echo', isn't it?

```
x='A B'
basename "$x"
```
can you give a better example?

**Comment by mulle-nat** [comment]
It's about the syntax of the backticks, the echo is just gratuitous. The example is fine as is.

**Comment by knirch** [comment]
Consider using echo "`basename "$xxx"`". The way I've learned is that quotes outside of command substitution will not treat quotes inside of it as part of the quoted string.. if I manage to make sense.

I was expecting your example to actually be treated as literal quotes sent off to basename, so I need to readjust my understanding. Regardless, shellcheck is happy with above example, and spaces are treated as expected.

**Comment by mulle-nat** [comment]
Nice, now I didn't think putting unescaped quotes into "` was possible. That may have come from writing to many simple parsers. :)

Thanks, I'll close this.

**Comment by deryni** [comment]
shellcheck is wrong there but only because the quoting inside backticks is odd. If that used `$(...)` substitution instead then shellcheck would be right that the escaped double quotes would not operate as word quotes and the warning would be correct. The difference is that backticks unescape the inner double quotes. Though I can't actually find the spec language that says that it should do that. A quick read makes it look like it shouldn't to me (or I'm misreading things).

**Comment by koalaman** [comment]
I see what happened. I read this part:

>backslash shall retain its literal meaning, except when followed by: '$', '`', or '\' (dollar sign, backquote, backslash). The search for the matching backquote shall be satisfied by the first backquote found without a preceding backslash

Which says that `\"` should be treated literally like shellcheck does.

However, it is in the context of this rule of Token Recognition:

>4. If the current character is backslash, single-quote, or double-quote ( '\', '", or ' )' and it is not quoted, it shall affect quoting for subsequent characters up to the end of the quoted text.

So it is indeed a valid parsing bug.

**Comment by mulle-nat** [comment]
OK, then I reopen it, so it don't get lost.

**Comment by koalaman** [comment]
Fixed in 1a52966

**Comment by HankB** [comment]
Is this the same issue?

```text
unxz -k --stdout "${image}" >"/tmp/$(basename -s .xz  \"${image}\")"
                                                        ^------^ SC2086 (info): Double quote to prevent globbing and word splitting.
```

Version 

```text
hbarta@olive:~/Programming/pmb$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
hbarta@olive:~/Programming/pmb$ 
```

I see that this was fixed quite a while ago but as I'm on Debian stable I'm not sure this fix is in the version I'm using. Also seems like it might be a slightly different issue as I'm using `$()` instead of back-ticks.

Thanks!

**Comment by ale5000-git** [comment]
@HankB
The most important advantage of using `$() ` instead of backticks is that the quotes do not have to be escaped.
So you code is wrong.
